app/gui/src/main/scala/org/bitcoins/gui/AppMenuBar.scala

```diff
@@ -35,8 +35,9 @@ private class FileMenu() {
         initialFileName = "bitcoin-s-backup.zip"
       }
       val chosenFile = fileChooser.showSaveDialog(null)
-      ConsoleCli.exec(ZipDataDir(chosenFile.toPath),
-                      GlobalData.consoleCliConfig)
+      if (chosenFile != null)
```

Comment:
So the best way to handle this in Scala is wrapping the potentially null thing in `Option`. This gives us types rather than `null` which circumvents the entire type system. Unfortunately other programming languages like Java has decided to use `null` rather than types to represent something that is not available.

```scala
val chosenFileOpt = Option(fileChoose.showSaveDialog(null))
chosenFileOpt match {
  case Some(choseFilen) => 
      ConsoleCli.exec(ZipDataDir(chosenFile.toPath),GlobalData.consoleCliConfig)
  case None => 
     //should we warn or something in this case?
```

Reply:
there's no need to log, this will happen if the user hit cancel instead of selecting a file